CMakeLists.txt

```diff
@@ -9,7 +9,12 @@ project(libtorrent
 	DESCRIPTION "Bittorrent library"
 	VERSION ${VER_MAJOR}.${VER_MINOR}.${VER_TINY}
 )
-set (SOVERSION "10")
+set (LIBVERSION_MAJOR 11)
+set (LIBVERSION_MINOR 0)
+set (LIBVERSION_PATCH 0)
```

Comment:
This seems it would introduce a lot of confusion. The version is 2.0.0. the SOVERISON may be something else, because that's there for ABI compatibility, but elevating it up to these names seem it would cause confusion

Reply:
Having the SOVERSION at 10 and the VERSION at 2.0.0 will be very confusing on linux.
If you check any other library, you'll see libfoo.1.2.3 linked to libfoo.so.1 linked to libfoo.so.

Instead, with things as they are now, libtorrent-rasterbar.so.10 links to libtorrent-rasterbar.so.2.0.0 which links to libtorrent-rasterbar.so, which can't really be used in a package.

I had listed major/minor/patch for clarity, but if you decide to use VERSION, major can be the SOVERSION and those other variables will be supplied directly by cmake (CMAKE_PROJECT_VERSION_MAJOR and so on).